stan/math/rev/mat/fun/softmax.hpp

```diff
@@ -4,81 +4,78 @@
 #include <stan/math/prim/arr/err/check_nonzero_size.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <stan/math/prim/mat/fun/softmax.hpp>
-#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/mat.hpp>
 #include <vector>
 
 namespace stan {
 namespace math {
 
 namespace {
-class softmax_elt_vari : public vari {
- private:
-  vari** alpha_;
-  const double* softmax_alpha_;
-  const int size_;  // array sizes
```

Comment:
Rather than doc this, how about using a variable like `N_alpha_` or even `size_alpha_`?  It's always better to name carefully than doc.